CONTRIBUTING.md

```diff
@@ -66,6 +67,112 @@ please ensure that:
+back-and-forth where the reviewer may ask a couple of questions about why things
+were done a particular way, and make suggestions for improvement.  You don't
+need to do everything suggested if you've got good reasons to disagree, but
+communicate that clearly and politely.
+
+If you're struggling with code review or a PR on Qiskit, you can ask for help in
+the `#qiskit-pr-help` channel on [the public Qiskit Slack](https://qisk.it/join-slack).
+
+Remember that the PR author is a human, not just a username!  It's ok to ask
+questions about the code, but don't be mean or rude about it even if you don't
+like it.
+
+#### What to focus on in review
+
+* Is everything in [the PR checklist](#pull-request-checklist) done?
+* Are any new public APIs easy to use, well documented, and consistent with
+  other parts of the Qiskit API?
+* Do any changes to the code have knock-on effects for other parts of Qiskit
+  that may be using them, or do they imply changes to the assumptions in
+  our data structures?
+* Are there any edge cases you can think of that the code might not handle well?
+* Is the code reasonably easy for you to understand?  This particular point is
+  tricky; the more you review code, the easier it will be for you to understand
+  more code, so don't worry about this as much if you're getting started.
+
+#### Writing review comments
+
+* Make concrete suggestions when you think something should be changed, but
+  remember that the author might have already thought about it and have a
+  reason.  Try "what do you think about us raising a `TypeError` here instead of
+  returning `None`?", rather than "you should raise an exception here".
+* Try to make each round of review thorough.  Don't add one or two comments on
+  one file, then come back a day later and add a couple of other unrelated
+  comments on a different file, and so on.  Try to review the whole PR
+  thoroughly in one go; it's easier to catch bugs like this, and less
+  frustrating to be on the other side of. If that's too much for you, consider
+  asking if the PR could be split into smaller independent chunks.
+* Try to keep the number of comments reasonable.  This depends on the size of
+  the PR, but remember that there's somebody who'll read all your comments, and
+  it can be demoralizing if you get a PR back and it's got 30 comments on from a
+  50-line change.  If you feel like you're putting too many comments on,
+  consider if you could group several of them into one theme, and ask them as a
+  more detailed question with a focus on only one part of the code.  Try not to
+  comment the same thing in many places.
+* Try to avoid saying "you do" in review comments, and instead try to
+  say things like "we do" even when talking about new code. It's not a big
+  change, yet it helps to make us think about Qiskit's code as something that we
+  all own and care about, and that we're all working together to make it better.
+
+#### Responding to review comments
```

Comment:
do you think it's worth adding a point to this section about respecting the expertise of core maintainers? Something about how it's ok to push back or question a review but at the end of the day if a maintainer is asking for a change you should respect and action their request if you want them to merge your work.

Reply:
I considered this when writing, and wasn't super sure whether I should say something like it or not.  I don't want to give the impression that maintainers are infallible, and we really would want maintainers to be able to back up change requests with convincing arguments when required (otherwise are we sure they're not pure opinion, which we want to avoid commenting on?), but I also do think that at some point yeah, the maintainer is the final say even if you don't fully buy their argument.

I can expand saying "the maintainer has the final say" (in nicer language), if that's something we all agree we should have?